server/data/restClient.ts

```diff
@@ -9,21 +9,20 @@ import type { ApiConfig } from '../config'
 import type { UnsanitisedError } from '../sanitisedError'
 import { restClientMetricsMiddleware } from './restClientMetricsMiddleware'
 
-interface GetRequest {
+interface BaseRequest {
   path?: string
   query?: object | string
   headers?: Record<string, string>
   responseType?: string
   raw?: boolean
+}
+
+interface Request extends BaseRequest {
```

Comment:
couldn't this handle404 be in the `BaseRequest`? Maybe there is a need for this additional level/class.

Reply:
the client only had 404 handling (i.e. swallowing a 404 error and returning a null instead of throwing an error) in the `get` method. do you think that we should have this 404 handling in other methods?